libvast/vast/system/run_reader.hpp

```diff
@@ -0,0 +1,101 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#ifndef VAST_SYSTEM_RUN_READER_HPP
+#define VAST_SYSTEM_RUN_READER_HPP
+
+#include <memory>
+#include <string>
+#include <string_view>
+
+#include <caf/scoped_actor.hpp>
+#include <caf/typed_actor.hpp>
+#include <caf/typed_event_based_actor.hpp>
+
+#include "vast/expression.hpp"
+#include "vast/logger.hpp"
+
+#include "vast/system/run_reader_base.hpp"
+#include "vast/system/run_reader_base.hpp"
+#include "vast/system/signal_monitor.hpp"
+#include "vast/system/source.hpp"
+#include "vast/system/tracker.hpp"
+
+#include "vast/concept/parseable/to.hpp"
+
+#include "vast/concept/parseable/vast/expression.hpp"
+#include "vast/concept/parseable/vast/schema.hpp"
+
+#include "vast/detail/make_io_stream.hpp"
+
+namespace vast::system {
+
+/// Default implementation for import sub-commands. Compatible with Bro and MRT
+/// formats.
+/// @relates application
+template <class Reader>
+class run_reader : public run_reader_base {
+public:
+  run_reader(command* parent, std::string_view name)
+      : run_reader_base(parent, name),
+        uds(false) {
+    this->add_opt("read,r", "path to input where to read events from", input);
+    this->add_opt("schema,s", "path to alternate schema", schema_file);
+    this->add_opt("uds,d", "treat -r as listening UNIX domain socket", uds);
+  }
+
+protected:
+  expected<caf::actor> make_source(caf::scoped_actor& self,
+                                   caf::message args) override {
+    CAF_LOG_TRACE(CAF_ARG(args));
+    auto in = detail::make_input_stream(input, uds);
+    if (!in)
+      return in.error();
+    Reader reader{std::move(*in)};
+    auto src = self->spawn(source<Reader>, std::move(reader));
```

Comment:
You could add a TODO comment here so that we remember to do the parsing of a user-provided schema and/or expression first, before spawning a source. (We talked about providing this information in the ctor.)